packages/airnode-node/src/evm/gas-prices.ts

```diff
@@ -14,7 +14,7 @@ export const parsePriorityFee = ({ value, unit }: PriorityFee) => ethers.utils.p
 const getLegacyGasPrice = async (options: FetchOptions): Promise<LogsData<GasTarget | null>> => {
   const { provider } = options;
 
-  const [err, gasPrice] = await go(provider.getGasPrice, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
+  const [err, gasPrice] = await go(() => provider.getGasPrice(), { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
```

Comment:
Mhh, I don't understand how this solves the issue - this looks like noop to me.

Reply:
We're learning ;)
The failure is to do with the context in which the function is called and that this function is a class-based function (relies on a field in the `this` object). The field in question is the RPC URL.
Most of our stuff is pure/functional/not class-based, but ethers is class based.
The automated tests didn't pick up on it because we mock the return value of the `getGasPrice` function. Only an E2E test against a node instance would have picked up on it *if* both an eip1559 and legacy scenario were used.

Reply:
Mhh, I'll need to lookup the `this` rules again...

Reply:
Yeah this was also really confusing to me. I still don't really get how `this` will be different between `provider.getGasPrice` and `() => provider.getGasPrice()` 🤔 

JS is a weird language 😄

Reply:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this

```
In strict mode, however, if the value of this is not set when entering an execution context, it remains as undefined, as shown in the following example:
```

The `this` in javascript is based on a dynamic scope, not lexical. E.g. this works:
```
'use strict';

function fn() {
  console.log(this.x);
}

const obj = { x: 'abc' };
obj.fn = fn;
// By calling the fn with "obj." the this is bound to the "obj"
obj.fn();
```